canonical/vanilla-framework
# feat: Implement content grid (`.grid-row` and `.grid-col`)

## Done

- Creates the new "content grid" - a grid with 4 columns on small and medium, and 8 on large. This is currently a smaller re-implementation of the grid component, per [team discussion](https://github.com/canonical/vanilla-framework/pull/5352#issuecomment-2393714486).
- Deprecated the "grid" component. Added code comments where reasonable to mark what styling needs to be adapted for the new grid.
- Added support for using this grid to a couple of components in cases where it is trivial.
   - Expanding table row (`.p-table--expanding (th, td).p-table__expanding-panel .grid-row`)
   - HR inside a grid (`.grid-row hr`)

Fixes [WD-14972](https://warthogs.atlassian.net/browse/WD-14972)

## QA

- Review [new grid](https://vanilla-framework-5368.demos.haus/docs/patterns/grid) docs
- Review [legacy grid](https://vanilla-framework-5368.demos.haus/docs/patterns/grid-legacy) docs
- Review [4.18.0 release notes](https://vanilla-framework-5368.demos.haus/docs/whats-new)
- Open [combined new grid example](https://vanilla-framework-5368.demos.haus/docs/examples/patterns/grid-8/combined?theme=light) and verify the examples look correct on all breakpoints.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

![image](https://github.com/user-attachments/assets/d0413fe0-7db0-41aa-9e26-29cc9cccada8)
![image](https://github.com/user-attachments/assets/f4a5c830-90b5-4774-bf99-b043aabc547d)
![image](https://github.com/user-attachments/assets/59f5693b-9e44-4c5b-9df6-9a0d889d3511)


[WD-14972]: https://warthogs.atlassian.net/browse/WD-14972?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
